src/vos/vos_obj.c

```diff
@@ -657,6 +657,7 @@ key_iter_fetch(struct vos_obj_iter *oiter, vos_iter_entry_t *ent,
 
 	ent->ie_epoch = epr.epr_hi;
 	ent->ie_punch = oiter->it_ilog_info.ii_next_punch;
+	ent->ie_obj_punch = oiter->it_obj->obj_ilog_info.ii_next_punch;
```

Comment:
@NiuYawei this is the key line of code here.   oiter->it_ilog_info refers to the incarnation log of the key whereas obj_ilog_info is the incarnation log of the object.   So this records the future object punch in the iterator result.

Reply:
ah, so this be used as punch epoch of current object? Since we do not use oi iteration during migration object enumeration, so we either get the obj punch epoch either from key enumeration or some other way. Thanks.

Reply:
correct.   When iterating the dkeys, this will tell you if the corresponding object is punched.  It's a bit redundant since it should be set for every dkey and you really only need to save it once